Logger/BaseLoggerMixins.cs

```diff
@@ -1,7 +1,71 @@
-﻿namespace Logger
+﻿using System;
+
+namespace Logger
 {
     public static class BaseLoggerMixins
     {
-        
+        public static void Error(this BaseLogger baseLogger, string message, params object[] args)
+        {
+            if (baseLogger is null)
+            {
+                throw new ArgumentNullException("baseLogger is null");
+            }
+
+            for (int i = 0; i < args.Length; i++)
+            {
+                string toBeReplaced = "{" + i + "}";
+                message = message.Replace(toBeReplaced, args[i].ToString());
+            }
+
```

Comment:
There is a much simpler way to do this using `string.Format`

```suggestion
string formattedMessage = string.Format(message, args);
```